tests/integration/metrics.test.js

```diff
@@ -202,4 +202,78 @@ function createMetricsTests(t) {
       t.end()
     })
   })
+
+  t.test('named query with fragment, query first', (t) => {
+    const { helper, serverUrl } = t.context
+
+    const expectedName = 'GetBookForLibrary'
+    const query = `query ${expectedName} {
+      library(branch: "downtown") {
+        books {
+          ... LibraryBook
+        }
+      }
+    }
+    fragment LibraryBook on Book {
+      title
+      author {
+        name
+      }
+    }`
+
+    const path = 'library.books.LibraryBook'
+
+    helper.agent.on('transactionFinished', () => {
+      const operationPart = `query/${expectedName}/${path}`
+
+      t.metrics([
+        `${OPERATION_PREFIX}/${operationPart}`,
+        `${RESOLVE_PREFIX}/library`,
+        `${RESOLVE_PREFIX}/books`,
+        `${RESOLVE_PREFIX}/author`
+      ])
+    })
+
+    executeQuery(serverUrl, query, (err) => {
+      t.error(err)
+      t.end()
+    })
+  })
+
+  t.test('named query with fragment, fragment first', (t) => {
+    const { helper, serverUrl } = t.context
+
+    const expectedName = 'GetBookForLibrary'
+    const query = `fragment LibraryBook on Book {
+      title
+      author {
+        name
+      }
+    }
+    query ${expectedName} {
+      library(branch: "downtown") {
+        books {
+          ... LibraryBook
+        }
+      }
+    }`
+
+    const path = 'library.books.LibraryBook'
```

Comment:
I think this naming is exactly what we want (not including the resolution of the fragment) but part of me is wondering if we should capture the details of the fragment in some way.

Maybe capturing fragments as attributes? 

Should prob wait for customer need/ask but a thought that occurred to me as what the fragment is selecting can have an impact on performance and may be nice not to have to dig back into code.

Reply:
I can see that but like you said I'd like to get asks from customers before tweaking things